### PR TITLE
setup: remove upper bound on kombu version

### PR DESCRIPTION
**Pull request by VMois**
Related https://github.com/reanahub/reana-server/pull/429

### CODE REVIEW

**Comment by VMois** [comment]
I guess, no need to change a version here. We can keep `0.8.2`.

**Comment by mvidalgarcia** [comment]
> I guess, no need to change a version here. We can keep `0.8.2`.

[0.8.2](https://pypi.org/project/reana-commons/0.8.2/) was already released. We have to release another version.

**Comment by VMois** [comment]
`PyGraphviz requires Python version 3.8 or later (3.7 detected)` looks like we need to pin pygraphviz at version <1.8 or drop support for 3.7 python (which is unlikely due to r-commons being used in r-client)

@mvidalgarcia

File: setup.py [hunk 1]
```diff
@@ -53,10 +53,11 @@ def get_snakemake_pkg(extras=""):
     "click>=7.0",
     "fs>=2.0",
     "jsonschema[format]>=3.0.1,<4.0.0",
-    "kombu>=4.6,<4.7",
+    "kombu>=4.6",
     "mock>=3.0,<4",
     "PyYAML>=5.1,<6.0",
     "Werkzeug>=0.14.1",
+    "pygraphviz<1.8",
```
Comment (VMois) [inline setup.py:60]: Not sure if this is the best decision. Maybe upgrading a package that brings `pygraphviz` would be a better idea, but, I guess this comes from `yadage` so unlikely there are any options.

**Comment by VMois** [comment]
I will need help with release as I have limited knowledge of the process. If we will release a new version of `r-commons`, it will unblock [`r-server` PR](https://github.com/reanahub/reana-server/pull/434)

File: setup.py [hunk 2]
```diff
@@ -32,7 +32,12 @@ def get_snakemake_pkg(extras=""):
     "docs": ["Sphinx>=1.4.4", "sphinx-rtd-theme>=0.1.9",],
     "tests": tests_require,
     "kubernetes": ["kubernetes>=11.0.0,<12.0.0",],
-    "yadage": ["adage~=0.10.1", "yadage~=0.20.1", "yadage-schemas~=0.10.6"],
+    "yadage": [
+        "adage~=0.10.1",
+        "yadage~=0.20.1",
+        "yadage-schemas~=0.10.6",
+        "pygraphviz<1.8",
```
Comment (mvidalgarcia) [inline setup.py:39]: [`snakemake[reports]` also installs this package](https://github.com/snakemake/snakemake/blob/5fabffbb4af8e9e122677e5adeaebf2d6bd0eeb3/setup.py#L81) so it probably should be at the `install_requires` level.

File: CHANGES.rst [hunk 3]
```diff
@@ -1,6 +1,11 @@
 Changes
 =======
 
+Version 0.8.3 (UNRELEASED)
+--------------------------
+
+- Changes installation to require kombu version 4.6 or higher.
```
Comment (mvidalgarcia) [inline CHANGES.rst:7]:
```suggestion
- Changes installation to require ``kombu`` version 4.6 or higher.
```

`kombu` version 4.6 or higher was partially required _in a way_. Perhaps it's more helpful to mention that the upper pin was removed.

File: setup.py (hunk 2 above)
Comment (VMois) [reply setup.py:39]: The only thing is that if nor `snakemake`, nor `yadage` are installed, `pygraphviz` still will be still installed.

Comment (mvidalgarcia) [reply setup.py:39]: Let's add it only as part of `snakemake_reports`.

Comment (mvidalgarcia) [reply setup.py:39]: Although, this is only part of the `reports` extra, and that's only being used in the workflow engine, that runs py38, so we should be safe without it...

**Review by mvidalgarcia — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (mvidalgarcia) [reply setup.py]: Don't we need the pin also in the `yadage` extra?

Comment (VMois) [reply setup.py]: `pygraphviz` in `yadage` is already pinned ([source](https://github.com/yadage/adage/blob/353c15cd97ff5150eff128f34cf1666c78826524/setup.py#L7)). CI for python 3.6 and 3.7 passing.

Comment (mvidalgarcia) [reply setup.py]: Ah good, in addition, that's the `viz` extra and we don't install it.

**Pull request merged**